junifer/preprocess/confounds/fmriprep_confound_remover.py

```diff
@@ -578,12 +679,38 @@ def preprocess(
                     }
                 }
             )
+
+        signal_clean_kwargs = {}
+        # Set up scrubbing mask if needed
+        if self.strategy.get("scrubbing", False):
+            motion_outline_regressors = self._get_scrub_regressors(
+                input["confounds"]["data"]
+            )
+            # Add regressors to confounds
+            confounds_df = pd.concat(
```

Comment:
Here you add the required variables for scrubbing to the df. Are they used later as regressors? Or they are not considered?

I fear that we might be regressing out this variables too.

Reply:
Motion outlier regressors are generated using the thresholds: https://github.com/nilearn/nilearn/blob/7a8cd0ee7161c4ea5ce4c2df248f0113890e3070/nilearn/interfaces/fmriprep/load_confounds_components.py#L222 and then added to the df: https://github.com/nilearn/nilearn/blob/7a8cd0ee7161c4ea5ce4c2df248f0113890e3070/nilearn/interfaces/fmriprep/load_confounds.py#L484 which then gets processed to generate the sample mask and the added motion outlier regressors are removed to return the proper confounds. This is leveraging nilearn's implementation so no code of ours should affect it.

Reply:
ok then.